src/test/java/org/prebid/server/it/PriceFloorsTest.java

```diff
@@ -0,0 +1,60 @@
+package org.prebid.server.it;
+
+import io.restassured.response.Response;
+import org.json.JSONException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.prebid.server.model.Endpoint;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static java.util.Collections.singletonList;
+
+@RunWith(SpringRunner.class)
+@TestPropertySource("floors-application.properties")
+public class PriceFloorsTest extends IntegrationTest {
+
+    @Test
+    public void auctionFloorsTest() throws IOException, JSONException {
+        // for both tests
+        // given
+        WIRE_MOCK_RULE.stubFor(get(urlPathEqualTo("/floors-provider"))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/floors/provided-floors.json"))));
+
+        // first run
+        // given
+        WIRE_MOCK_RULE.stubFor(post(urlPathEqualTo("/generic-exchange"))
+                .withId(UUID.fromString("de0755a6-9878-4c61-a841-b725a02099bb"))
```

Comment:
Instead of withId, is it possible to use `inScenario`, `whenScenarioStateIs` and `willSetStateTo` methods?